auth/feature-tests/functional/driver/auth.driver.ts

```diff
@@ -1,52 +1,80 @@
+  access_token: string;
+  id_token: string;
+  refresh_token: string;
 }
 
 export class AuthDriver {
-  private cognitoClient: CognitoIdentityProviderClient;
   private clientId: string;
+  private authUrl: string;
+  private redirectUri: string;
 
-  constructor(clientId: string, region: string) {
+  constructor(clientId: string, authUrl: string) {
     this.clientId = clientId;
-    this.cognitoClient = new CognitoIdentityProviderClient({ region });
+    this.authUrl = authUrl
+    // must be a valid server and redirect uri to allow puppeteer to intercept the request
+    this.redirectUri = "https://d84l1y8p4kdic.cloudfront.net"
   }
 
-  /**
-   * Initiates the authentication process as a client application.
-   *
-   * @returns A Promise that resolves to the InitiateAuthCommandOutput from Cognito.
-   * @throws Error if the initiation process fails.
-   */
-  async initiateAuth({
-    username, 
-    password,
-    attestationToken
-}: InitiateAuthInput): Promise<InitiateAuthCommandOutput> {
-    const authParams: InitiateAuthRequest = {
-      AuthFlow: "USER_PASSWORD_AUTH", 
-      ClientId: this.clientId,
-      AuthParameters: {
-        USERNAME: username,
-        PASSWORD: password
+  async loginAndGetCode(input: LoginUserInput) {
+    // Use puppeteer or simulate GET to /authorize
+    const authorizeUrl = `${this.authUrl}/oauth2/authorize?` + querystring.stringify({
+        response_type: 'code',
+        client_id: this.clientId,
+        redirect_uri: this.redirectUri,
+        scope: 'email openid',
+        state: 'xyz123' // (could be random)
+      });
+
+    const browser = await puppeteer.launch({ headless: true });
+    const page = await browser.newPage();
+  
+    await page.goto(authorizeUrl);
+
+    await page.type('input[name="username"]', input.username);
+    await page.type('input[name="password"]', input.password);
+    await page.click('input[name="signInSubmitButton"]');
+  
+    await page.waitForNavigation();
+  
+    const redirectedUrl = page.url();  
+  
+    const url = new URL(redirectedUrl);
+    const code = url.searchParams.get('code');
+
+    if (!code) throw new Error('No auth code found in redirect URL');
+    await browser.close();
+
+    return code;
+  }
+
+  async exchangeCodeForTokens(code: string): Promise<TokenExchangeResponse> {
+    const response = await fetch(`${this.authUrl}/oauth2/token`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
       },
-      ClientMetadata: {
-        attestationToken
-      }
-    };
-
-    try {
-      const command = new InitiateAuthCommand(authParams);
-      const response = await this.cognitoClient.send(command);
-      return response;
-    } catch (error: any) {
-      console.error("Error initiating authentication:", error);
-      throw new Error(`Failed to initiate authentication: ${error.message}`);
+      body: querystring.stringify({
+        grant_type: 'authorization_code',
```

Comment:
Note this can be changed later when PKCE is enabled. This test presumes an app client is created - which is WIP